### ISSUE TITLE
[Small Feature Suggestion] Adding WASD as options for keyboard movement controls

### ISSUE THREAD

**Issue by austintian03** [Tuxemon/Tuxemon#721]
It's not the most important thing, but just an idea cause it's a familiar movement scheme for a lot of people. Plus with WASD it's less awkward to move and interact with the game using ENTER. So it definitely wouldn't hurt to add, especially since it doesn't seem too difficult to implement.

How I thought to implement it was to add these lines of code to tuxemon/core/platform/platform_pygame/events.py:

```python
class PygameKeyboardInput(...) { 
     default_input_map = {
        . . .
        pg.K_w: buttons.UP,
        pg.K_s: buttons.DOWN,
        pg.K_a: buttons.LEFT,
        pg.K_d: buttons.RIGHT,
        . . . 
     }
}
```

So this is just mapping the 4 keys WASD to the respective buttons for directions, but I don't know if it's bad practice to add 2 sets of movement configurations in the *default_input_map*. 

Please let me know if there's a better way to do this. Thanks!

**Comment by Qiangong2** [comment]
Could it be possible to have a setting in the tuxemon.cfg? That way it can stay the default, but people can change if they want.

Also, I just use left enter when I need to

**Comment by austintian03** [comment]
I'm not too sure, but I can look into it if I can find tuxemon.cfg lol.
Also, what do you mean by left enter? I don't believe I have a left enter on my laptop keyboard or my desktop keyboard.

**Comment by kerizane** [comment]
tuxemon.cfg is generated by tuxemon/core/config.py

**Comment by bitcraft** [comment]
Being able to change the default map is partially completed.  In order to complete it, the following are needed:

* Define how the config file would look.  maybe like the following?
```
[controls]
player_up = w
player_down = s
...
```
* add the code to `config.py` to read the controls and create the control map dictionary
* set `Config.keyboard_button_map` to this dictionary

All of the work is part of reading the config.  Remapping the controls can be done once the config is changed.  This final dictionary would look something like this:
```python
    {
        pg.K_w: buttons.UP,
        pg.K_s: buttons.DOWN,
        pg.K_a: buttons.LEFT,
        pg.K_d: buttons.RIGHT,
        pg.K_RETURN: buttons.A,
        pg.K_RSHIFT: buttons.B,
        pg.K_LSHIFT: buttons.B,
        pg.K_ESCAPE: buttons.BACK,
        pg.K_BACKSPACE: events.BACKSPACE,
        None: events.UNICODE,
    }
```

I did most of this a while back, but didn't finish the part to read the config file.  Remapping the controls works, and I just tested this by setting `Control.keyboard_button_map` to something else.

**Comment by austintian03** [comment]
So to clarify, as this is my understanding of it, this solution is done by manually editing `tuxemon.cfg`?

**Comment by bitcraft** [comment]
the ability to change it in the config file is not coded yet.  you can however, edit the source file (`default_key_map`; same where you linked it before) and it will work fine, but for you only.  You can use my post above as a template for the changes.  It should work fine.

**Comment by austintian03** [comment]
Yeah, I had already tested that change in the 'default_key_map' and it works, but it's definitely not elegant as one can use both WASD and Arrow keys to move around (though that's not necessarily a problem as it lets people choose their movement scheme without having to edit any external files).

Alright, can I claim this issue of enabling editing of the default keys in the config file then?

**Comment by bitcraft** [comment]
Sure, feel free to tackle the problem.  If you need pointers you can ask here, but the discord may be faster.  As for using both arrow keys and WASD,
```
    {
        ....
        pg.K_RETURN: buttons.A,
        pg.K_RSHIFT: buttons.B,
        pg.K_LSHIFT: buttons.B,
        ....
    }
```
Notice that `pg.K_RSHIFT` and `pg.K_LSHIFT` are used for `buttons.B`, so you can use both arrow keys and WASD in the same way so they can both be used.  I do think there may be a problem with the name entry screen since typing characters WASD will *maybe* also move the cursor when typing on the keyboard.  Please test this and open an issue if it happens.

**Comment by bitcraft** [comment]
Implemented in #747
